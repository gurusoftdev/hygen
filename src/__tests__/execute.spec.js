const execute = require('../execute')
const ftest = require('../test/ftest')
const fs = require('fs')

describe('execute', () => {
  ftest(
    'works',
    {
      app: {}
    },
    () => {
      execute(
        'app',
        [{ attributes: { to: 'workers/foobar.js' }, body: 'hello js!' }],
        () => {
          throw new Error('prompt!')
        },
        {},
        { logger: console }
      )

      expect(
        fs.readFileSync('app/workers/foobar.js').toString()
      ).toMatchSnapshot()
    }
  )

  ftest(
    'skips, prompt=n',
    {
      app: { workers: { 'foobar.js': 'foobar' } }
    },
    () => {
      execute(
        'app',
        [{ attributes: { to: 'workers/foobar.js' }, body: 'hello js!' }],
        () => {
          return 'n'
        },
        {},
        { logger: console }
      )
    }
  )

  ftest(
    'overwrite, prompt=y',
    {
      app: { workers: { 'foobar.js': 'foobar' } }
    },
    () => {
      let flagged = false
      execute(
        'app',
        [{ attributes: { to: 'workers/foobar.js' }, body: 'hello js!' }],
        () => {
          flagged = true
          return 'y'
        },
        {},
        { logger: console }
      )
      expect(flagged).toBe(true)
    }
  )
})
