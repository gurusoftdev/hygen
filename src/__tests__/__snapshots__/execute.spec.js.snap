// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute works 1`] = `"hello js!"`;
